### PR TITLE
Print `as` instead of `:` for `babel-ts` parser

### PR DESCRIPTION
**Pull request by HosokawaR**
## Description

fix: #10635

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/print/misc.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ function printTypeAnnotation(path, options, print) {
 
   const parentNode = path.getParentNode();
 
+  if (parentNode.type === "ArrayPattern" && options.parser === "babel-ts") {
+    return [" as ", print("typeAnnotation")];
+  }
```
Comment (sosukesuzuki) [inline src/language-js/print/misc.js:60]: Ideally we should fix this in babel-parser side, but this seems to be work well as temporary fix.

**Review by sosukesuzuki — APPROVED** [review]

File: src/language-js/print/misc.js (hunk 1 above)
Comment (fisker) [reply src/language-js/print/misc.js:60]: Has this been reported?

File: changelog_unreleased/typescript/12706.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#### Print `as` instead of `:` for `babel-ts` parser (#12706 by @HosokawaR)
+
+<!-- Optional description if it makes sense. -->
+
```
Comment (fisker) [inline changelog_unreleased/typescript/12706.md:4]:
```suggestion
```

File: src/language-js/print/misc.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-js/print/misc.js:60]: done https://github.com/babel/babel/issues/14498

**Review by fisker — APPROVED** [review]

**Pull request merged**